ddspipe_participants/src/cpp/reader/rtps/CommonReader.cpp

```diff
@@ -310,51 +307,71 @@ fastrtps::rtps::ReaderAttributes CommonReader::reckon_reader_attributes_(
     // Set if topic has key
     if (topic.topic_qos.keyed)
     {
-        att.endpoint.topicKind = eprosima::fastrtps::rtps::WITH_KEY;
+        att.endpoint.topicKind = eprosima::fastdds::rtps::WITH_KEY;
 
         // If the topic has a key, request inline qos (containing the instance handle)
-        att.expectsInlineQos = true;
+        att.expects_inline_qos = true;
     }
     else
     {
-        att.endpoint.topicKind = eprosima::fastrtps::rtps::NO_KEY;
+        att.endpoint.topicKind = eprosima::fastdds::rtps::NO_KEY;
     }
 
     // Ownership and Partitions are not part of RTPS, thus they are set in properties
 
     return att;
 }
 
-fastrtps::TopicAttributes CommonReader::reckon_topic_attributes_(
+fastdds::rtps::TopicDescription CommonReader::reckon_topic_description_(
         const core::types::DdsTopic& topic) noexcept
 {
-    fastrtps::TopicAttributes att;
+    fastdds::rtps::TopicDescription topic_description;
 
-    // Set if topic has key
-    if (topic.topic_qos.keyed)
-    {
-        att.topicKind = eprosima::fastrtps::rtps::WITH_KEY;
-    }
-    else
-    {
-        att.topicKind = eprosima::fastrtps::rtps::NO_KEY;
-    }
+    topic_description.type_name = topic.m_topic_name;
+    topic_description.topic_name = topic.type_name;
 
-    // Set Topic attributes
-    att.topicName = topic.m_topic_name;
-    att.topicDataType = topic.type_name;
-
-    // Set Topic history attributes
-    att.historyQos.kind = eprosima::fastdds::dds::HistoryQosPolicyKind::KEEP_LAST_HISTORY_QOS;
-    att.historyQos.depth = topic.topic_qos.history_depth;
-
-    return att;
+    return topic_description;
```

Comment:
Missing to assign type_information.

Reply:
I think you forgot to only look for the complete type object in CommonWriter. Anyway, I believe we should implement some logic so we transmit only what is registered, i.e.
1. Look for the pair
2. If fails, look for complete only
3. If fails, look for minimal only 

This is obviously not efficient, but I don't think Fast-DDS API will be extended just to cover this case.